cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/AbstractClusterOperations.java

```diff
@@ -258,6 +259,7 @@ public final void reconcile(String namespace, String name) {
                                 }
                                 lock.release();
                                 log.debug("Lock {} released", lockName);
+                                fut.complete();
```

Comment:
I think this is wrong. The timer should restart after the reconciliation was done. Not after the action was taken. A rolling update of some cluster might take ages to complete. You do not want to block the reconciliation for all clusters during this time. The Future has to complete immediately when this function completes.

Reply:
The fact that the CC could watch more clusters in the same namespace doesn't want to find space in my mind ! You are right ;)